lib/improver/utilities/temporal_interpolation.py

```diff
@@ -0,0 +1,419 @@
+
+        Args:
+            cube (iris.cube.Cube):
+                cube containing diagnostic cube valid at the beginning
+                of the period and cube valid at the end of the period
+            interpoldated_cube (iris.cube.Cube):
+                cube containing Linear interpolation of
+                cube at times in time_list.
+        Returns:
+            interpolated_cubes (iris.cube.CubeList):
+                A list of cubes interpolated to the desired times.
+
+        """
+
+        interpolated_cubes = iris.cube.CubeList()
+        (lats, lons) = self.calc_lats_lons(cube)
+        prev_data = cube[0].data
+        next_data = cube[1].data
+        dtvals = iris_time_to_datetime(cube.coord('time'))
+        dtval_prev = dtvals[0]
+        sin_phi_prev = self.calc_sin_phi(dtval_prev, lats, lons)
+        dtval_next = dtvals[1]
+        sin_phi_next = self.calc_sin_phi(dtval_next, lats, lons)
+        diff_step = (dtval_next - dtval_prev).seconds
+
+        for single_time in interpolated_cube.slices_over('time'):
+            # new_cubelist = iris.cube.CubeList()
+            dtval_interp = iris_time_to_datetime(single_time.coord('time'))[0]
+            sin_phi_interp = self.calc_sin_phi(dtval_interp, lats, lons)
+            diff_interp = (dtval_interp - dtval_prev).seconds
+            single_time.data[::] = 0.0
+            sun_up = np.where(sin_phi_interp > 0.0)
+            if len(single_time.shape) > 2:
+                prevv = (
+                    prev_data[::, sun_up[0], sun_up[1]]/sin_phi_prev[sun_up])
+                nextv = (
+                    next_data[::, sun_up[0], sun_up[1]]/sin_phi_next[sun_up])
+                single_time.data[::, sun_up[0], sun_up[1]] = (
+                    sin_phi_interp[sun_up] *
+                    (prevv + (nextv - prevv) * (diff_interp/diff_step)))
+            else:
+                prevv = prev_data[sun_up]/sin_phi_prev[sun_up]
+                nextv = next_data[sun_up]/sin_phi_next[sun_up]
+                single_time.data[sun_up] = (sin_phi_interp[sun_up] *
+                                            (prevv + (nextv - prevv)
+                                             * (diff_interp/diff_step)))
+
+            interpolated_cubes.append(single_time)
+        return interpolated_cubes
+
+    @staticmethod
+    def daynight_interpolate(cube, interpolated_cube):
+        """
+        Interpolate solar radiation parameter which are zero if the
+        sun is below the horizon.
+
+        Args:
+            cube (iris.cube.Cube):
+                cube containing diagnostic cube valid at the beginning
+                of the period and cube valid at the end of the period
+            interpoldated_cube (iris.cube.Cube):
+                cube containing Linear interpolation of
+                cube at times in time_list.
+
+        Returns:
+            interpolated_cubes (iris.cube.CubeList):
+                A list of cubes interpolated to the desired times.
```

Comment:
I don't think in this case you use `cube` at all.  In fact I believe you can just modify `interpolated_cube` in place, without needing to return a cube list here.  But if you want to keep the functions of the same structure please could you address the same questions as for `solar_interpolate` above.

Reply:
Yes cube is not used. Will remove it from the call.